src/quicr_client_raw_session.cpp

```diff
@@ -218,6 +241,119 @@ ClientRawSession::connecting() const
+                        + std::chrono::duration_cast<std::chrono::system_clock::duration>(data_sample->sample_time - std::chrono::steady_clock::now());
+
+        auto& data_measurement = data_ns_it->second;
+        data_measurement
+          .SetTime(tp)
+          .SetMetric("enqueued_objs", data_sample->quic_sample->enqueued_objs)
+          .SetMetric("tx_queue_size_min", data_sample->quic_sample->tx_queue_size.min)
+          .SetMetric("tx_queue_size_max", data_sample->quic_sample->tx_queue_size.max)
+          .SetMetric("tx_queue_size_avg", data_sample->quic_sample->tx_queue_size.avg)
+          .SetMetric("rx_stream_bytes", data_sample->quic_sample->rx_stream_bytes)
+          .SetMetric("rx_stream_cb", data_sample->quic_sample->rx_stream_cb)
+          .SetMetric("tx_dgrams", data_sample->quic_sample->tx_dgrams)
+          .SetMetric("tx_dgrams_bytes", data_sample->quic_sample->tx_dgrams_bytes)
+          .SetMetric("tx_stream_objs", data_sample->quic_sample->tx_stream_objects)
+          .SetMetric("tx_stream_bytes", data_sample->quic_sample->tx_stream_bytes)
+          .SetMetric("tx_buffer_drops", data_sample->quic_sample->tx_buffer_drops)
+          .SetMetric("tx_delayed_callback", data_sample->quic_sample->tx_delayed_callback)
+          .SetMetric("tx_queue_discards", data_sample->quic_sample->tx_queue_discards)
+          .SetMetric("tx_queue_expired", data_sample->quic_sample->tx_queue_expired)
+          .SetMetric("tx_reset_wait", data_sample->quic_sample->tx_reset_wait)
+          .SetMetric("tx_stream_cb", data_sample->quic_sample->tx_stream_cb)
+          .SetMetric("tx_callback_ms_min", data_sample->quic_sample->tx_callback_ms.min)
+          .SetMetric("tx_callback_ms_max", data_sample->quic_sample->tx_callback_ms.max)
+          .SetMetric("tx_callback_ms_avg", data_sample->quic_sample->tx_callback_ms.avg)
+          .SetMetric("tx_object_duration_us_min", data_sample->quic_sample->tx_object_duration_us.min)
+          .SetMetric("tx_object_duration_us_max", data_sample->quic_sample->tx_object_duration_us.max)
+          .SetMetric("tx_object_duration_us_avg", data_sample->quic_sample->tx_object_duration_us.avg);
+
+        publishMeasurement(data_measurement);
+      }
+    }
+    LOGGER_INFO(logger, "Finished metrics thread");
+  });
+}
+
+void
+ClientRawSession::publishMeasurement(const Measurement& measurement)
+{
+  if (!metrics_namespace) return;
+
+  const json measurement_json = measurement;
+  const std::string measurement_str = measurement_json.dump();
+  quicr::bytes measurement_bytes(measurement_str.begin(), measurement_str.end());
+
+  const auto start_time = std::chrono::time_point_cast<std::chrono::microseconds>(std::chrono::steady_clock::now());
+  std::vector<qtransport::MethodTraceItem> trace;
+  trace.reserve(10);
+  trace.push_back({"libquicr:publishMetrics:begin", start_time});
+
+  publishNamedObject(*metrics_namespace, 31, 500, std::move(measurement_bytes), std::move(trace));
+}
```

Comment:
looks like we are publishing without subscription ?
Also we are not going through intent before publishing ..

Reply:
Metrics adapter is always subscribed to the server (so long as it's running).

We send a publish intent when we receive a connection response. When we receive a publish intent response we start the running the metrics thread.

As for sending individual metrics through the API, this follows the same logic as publishNamedObject; You can try publishing without an intent, but it won't have any effect if you haven't sent a publish intent first.